src/sandbox/enviroments/twenty_forty_eight/env.py

```diff
@@ -0,0 +1,66 @@
+from typing import Optional, Union, Tuple
+from numpy.typing import NDArray
+import gym
+import numpy as np
+import random
+
+from sandbox.enviroments.twenty_forty_eight.game.game import TwentyFortyEightGame
+from sandbox.enviroments.twenty_forty_eight.game.state import TwentyFortyEightState
+from sandbox.enviroments.twenty_forty_eight.game.heuristic import TwentyFortyEightHeuristic
+from sandbox.enviroments.twenty_forty_eight.game.action import TwentyFortyEightPlayerAction, Direction
+
+
+class TwentyFortyEightEnv(gym.Env):
+    def __init__(self, invalid_move_threshold=16, invalid_move_percentage=0.1):
+        self.game = TwentyFortyEightGame()
+        self.heuristic = TwentyFortyEightHeuristic(self.game)
+        self.state = self.game.initial_game_state()
+        self.observation_space = gym.spaces.Box(0, 2048, (self.game.board_dim*self.game.board_dim, ), dtype=np.uint32)
```

Comment:
In 2048 game there are only 11 possible numbers 2^1, 2^2, ..., 2^11, therefore it would be better (especially for tabular policies) to store only an exponent.